test/tools/yulInterpreter/EWasmBuiltinInterpreter.cpp

```diff
@@ -0,0 +1,376 @@
+		arg.emplace_back(uint64_t(a & uint64_t(-1)));
+
+	if (_fun == "datasize"_yulstring)
+		return u256(keccak256(h256(_arguments.at(0)))) & 0xfff;
+	else if (_fun == "dataoffset"_yulstring)
+		return u256(keccak256(h256(_arguments.at(0) + 2))) & 0xfff;
+	else if (_fun == "datacopy"_yulstring)
+	{
+		// This is identical to codecopy.
+		if (accessMemory(_arguments.at(0), _arguments.at(2)))
+			copyZeroExtended(
+				m_state.memory,
+				m_state.code,
+				size_t(_arguments.at(0)),
+				size_t(_arguments.at(1) & size_t(-1)),
+				size_t(_arguments.at(2))
+			);
+		return 0;
+	}
+	else if (_fun == "drop"_yulstring)
+		return {};
+	else if (_fun == "unreachable"_yulstring)
+		throw ExplicitlyTerminated();
+	else if (_fun == "i64.add"_yulstring)
+		return arg[0] + arg[1];
+	else if (_fun == "i64.sub"_yulstring)
+		return arg[0] - arg[1];
+	else if (_fun == "i64.mul"_yulstring)
+		return arg[0] * arg[1];
+	else if (_fun == "i64.div_u"_yulstring)
+	{
+		if (arg[1] == 0)
+			throw ExplicitlyTerminated();
+		else
+			return arg[0] / arg[1];
+	}
+	else if (_fun == "i64.rem_u"_yulstring)
+	{
+		if (arg[1] == 0)
+			throw ExplicitlyTerminated();
+		else
+			return arg[0] % arg[1];
+	}
+	else if (_fun == "i64.and"_yulstring)
+		return arg[0] & arg[1];
+	else if (_fun == "i64.or"_yulstring)
+		return arg[0] | arg[1];
+	else if (_fun == "i64.xor"_yulstring)
+		return arg[0] ^ arg[1];
+	else if (_fun == "i64.shl"_yulstring)
+		return arg[0] << arg[1];
+	else if (_fun == "i64.shr_u"_yulstring)
+		return arg[0] >> arg[1];
+	else if (_fun == "i64.eq"_yulstring)
+		return arg[0] == arg[1] ? 1 : 0;
+	else if (_fun == "i64.ne"_yulstring)
+		return arg[0] != arg[1] ? 1 : 0;
+	else if (_fun == "i64.eqz"_yulstring)
+		return arg[0] == 0 ? 1 : 0;
+	else if (_fun == "i64.lt_u"_yulstring)
+		return arg[0] < arg[1] ? 1 : 0;
+	else if (_fun == "i64.gt_u"_yulstring)
+		return arg[0] > arg[1] ? 1 : 0;
+	else if (_fun == "i64.le_u"_yulstring)
+		return arg[0] <= arg[1] ? 1 : 0;
+	else if (_fun == "i64.ge_u"_yulstring)
+		return arg[0] >= arg[1] ? 1 : 0;
+	else if (_fun == "i64.store"_yulstring)
+	{
+		accessMemory(arg[0], 8);
+		writeMemoryWord(arg[0], arg[1]);
+		return 0;
+	}
+	else if (_fun == "i64.load"_yulstring)
+	{
+		accessMemory(arg[0], 8);
+		return readMemoryWord(arg[0]);
+	}
+	else if (_fun == "eth.getAddress"_yulstring)
+		return writeAddress(arg[0], m_state.address);
+	else if (_fun == "eth.getExternalBalance"_yulstring)
+		// TODO this does not read the address, but is consistent with
+		// EVM interpreter implementation.
+		// If we take the address into account, this needs to use readAddress.
+		return writeU128(arg[0], m_state.balance);
+	else if (_fun == "eth.getBlockHash"_yulstring)
+	{
+		if (arg[0] >= m_state.blockNumber || arg[0] + 256 < m_state.blockNumber)
+			return 1;
+		else
+			return writeU256(arg[1], 0xaaaaaaaa + u256(arg[0] - m_state.blockNumber - 256));
+	}
+	else if (_fun == "eth.call"_yulstring)
+	{
+		// TODO read args from memory
```

Comment:
`TODO` later or in this PR?

Reply:
It's not clear what to do with the read args, because the evm interpreter also does not do anything with it.